### PR TITLE
CAS-177: Integrate Leaderboard API with front-end

### PR DESCRIPTION
**Pull request by jbluks**
**Ticket:** [CAS-177](https://linear.app/dappercollectives/issue/CAS-177/add-leaderboard-score-to-community-member-page)

**Description**

Connects Leaderboard API with the Leaderboard component on the community page

Screen Shot 2022-07-12 at 11 16 41 AM

NOTE: Added a pattern to separate the API logic from the useQuery hook. This promotes reusability of the api logic, and simplifies the fetching hooks so that the only logic relates to the hook, and not to the fetching.

### CODE REVIEW

File: frontend/packages/client/src/api/leaderboard.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  `${COMMUNITIES_URL}/${communityId}/leaderboard?count=${LEADERBOARD_SIZE}`;
+
+const fetchLeaderBoard = async (communityId) => {
+  try {
+    const response = await fetch(getLeaderBoardUrl(communityId));
+    return await checkResponse(response);
+  } catch (err) {
+    throw err;
+  }
+};
```
Comment (germanurrus) [inline frontend/packages/client/src/api/leaderboard.js:15]: I like this approach of adding this in a separate file!

What do you think about making a variable `pageSize` a param to the Url then if that changes on the hook it's already prepared

```suggestion
const LEADERBOARD_SIZE = 10;
const getLeaderBoardUrl = (communityId, pageSize) =>
  `${COMMUNITIES_URL}/${communityId}/leaderboard?count=${pageSize}`;

const fetchLeaderBoard = async (communityId, pageSize = LEADERBOARD_SIZE ) => {
  try {
    const response = await fetch(getLeaderBoardUrl(communityId, pageSize));
    return await checkResponse(response);
  } catch (err) {
    throw err;
  }
};
```

File: frontend/packages/client/src/api/leaderboard.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { checkResponse } from '../utils';
+import { COMMUNITIES_URL } from './constants';
```
Comment (germanurrus) [inline frontend/packages/client/src/api/leaderboard.js:2]: looks like there's a double `/` . Here and then on line 6 it's adding another `/` Causing a fetch error.

File: frontend/packages/client/src/api/leaderboard.js (hunk 1 above)
Comment (jbluks) [reply frontend/packages/client/src/api/leaderboard.js:15]: Oh yes - that is a better approach.

File: frontend/packages/client/src/api/leaderboard.js (hunk 2 above)
Comment (jbluks) [reply frontend/packages/client/src/api/leaderboard.js:2]: Fixed.

File: frontend/packages/client/src/api/constants.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export const API_BASE_URL = process.env.REACT_APP_BACK_END_SERVER_API;
+
+export const COMMUNITIES_URL = `${API_BASE_URL}/communities/`;
```
[the remark this answers is not in the document]
Comment (jbluks) [reply frontend/packages/client/src/api/constants.js]: I added this as a potential new pattern, so we don't have process.env.REACT_APP... all over the app. The idea is that base URLS are built in here and used by any API call that needs them. Going forward, all api fetching/updating will be defined in this directory and use these constants. For example the COMMUNITIES_URL will be used by all queries to that end point.

File: frontend/packages/client/src/hooks/useLeaderBoard.js [hunk 4]
```diff
@@ -1,70 +1,29 @@
...
+    isError,
+    error,
+    data: {
+      leaderBoard: data?.data ?? [],
+      currentUser: {
+        addr: 'joshprin...',
+        score: '123 $XYZ',
+        index: 122,
+      },
+    },
```
[the remark this answers is not in the document]
Comment (jbluks) [reply frontend/packages/client/src/hooks/useLeaderBoard.js:27]: Yea - I will follow up with Product on this. If we make a change, will create a new ticket for it.

**Review by germanurrus — COMMENTED** [review]
Looks good 👍 (FE side) , thanks for the updates! just left some comments

File: frontend/packages/client/src/hooks/useLeaderBoard.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (jbluks) [reply frontend/packages/client/src/hooks/useLeaderBoard.js:27]: Followed up with Jose and it is expected that we show the current user and their score. I have created a ticket to add this feature: [CAS-129](https://linear.app/dappercollectives/issue/CAS-229/add-current-user-to-leaderboard)

File: frontend/packages/client/src/pages/Community.js [hunk 5]
```diff
@@ -185,7 +185,7 @@ export default function Community() {
 
   // these two fields should be coming from backend as configuration
   const showPulse = false;
-  const showLeaderBoard = false;
+  const showLeaderBoard = true;
```
Comment (CJamesKeller) [inline frontend/packages/client/src/pages/Community.js:188]: Do we have a ticket to grab this from the backend? If not, let's add one.

**Review by CJamesKeller — APPROVED** [review]

File: frontend/packages/client/src/components/LeaderBoard.js [hunk 6]
```diff
@@ -74,7 +77,7 @@ export default function LeaderBoard({ onClickViewMore = () => {} } = {}) {
             })}
```
[the remark this answers is not in the document]
Comment (germanurrus) [reply frontend/packages/client/src/components/LeaderBoard.js:77]: can you add the classes: `is-flex flex-1 is-justify-content-center has-text-weight-bold pr-2` to make it centered, it's looking like this:
Screen Shot 2022-07-14 at 12 32 21

**Review by germanurrus — APPROVED** [review]
Looks good, thanks @jbluks for the changes!

**Review by rvmelkonian — APPROVED** [review]
looks good, nice with the tests!

**Pull request merged**